clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/provider/agent/KubernetesInstanceCachingAgent.groovy

```diff
@@ -83,25 +67,28 @@ class KubernetesInstanceCachingAgent implements CachingAgent, AccountAware {
 
     Map<String, MutableCacheData> cachedInstances = MutableCacheData.mutableCacheMap()
 
-    def podEvents = [:]
+    Map<String, Map<String, List<Event>>>  podEvents = [:].withDefault { _ -> [:] }
     try {
-      podEvents = credentials.apiAdaptor.getEvents(namespace, "Pod")
+      namespaces.each { String namespace ->
```

Comment:
Might be better to cache the results of `getNamespaces()` instead of calling it multiple times.